### ISSUE TITLE
SentenceBert Fit-Method: Capture loss, Gradient Accumulation and Multi-GPU

### ISSUE THREAD

**Issue by patrick-vtc** [UKPLab/sentence-transformers#1340]
Hey there, thanks for your paper, really enjoyed reading it! I have a few questions about your trainer (fit-implementation). How is it possible to:

- log the train and val. loss, i.e. in a diagram. In your implementation, I cannot find a way to pass back the loss, in a callback or read it from a generated artifacts (checkpoints, results). How to deal with that? I guess it would be sufficient to pass the loss as callback parameter besides score, epochs and steps
- do gradient accumulation for training really large models
- do Multi-GPU training, torch.nn.DataParallel(model.to(device)) does not seem to work

Best regards and Merry X-Mas
Patrick

**Comment by itsmilishah** [comment]
You probably have a solution by now but just for others visiting this - I created my own loss class that keeps track of the losses it has calculated for every batch. And I created a callback object that has access to the loss object. So I can log the loss everytime the callback is called. And the loss class also maintains a running sum of the losses and the number of samples, so at the end of each epoch I calculate the average loss of the entire batch and then reset the running sum and number of samples.
